commitlint/helpers.ts

```diff
@@ -98,6 +98,11 @@ export abstract class Helpers {
         return isLowerCase && !isUpperCase;
     }
 
+    public static isEmptyFooterReference(line: string) {
+        Helpers.assertLine(line);
+        return line[0] === "[" && line.indexOf("]") > 0 && line.trim().length === 3;
```

Comment:
@parhamsaremi hardcoding 3 here is still wrong, not sure if you understood my previous comment

Reply:
hmmm right. I understand now. I'll check with regex to see if it's only footer ref number.

Reply:
I don't think using regex here is worth it, just check if the string last char before EOL (trimmed) is `]`

Reply:
:+1:

Reply:
In the end I just check if the indexof "]" is length-1

Reply:
I think I have to use regex to catch this :thinking:  do you agree? @knocte

Reply:
I changed it to regex and also there was some sanity check errors that I addressed. let me know if you still think I shouldn't use regex and checking last index to be "]" is ok.

Reply:
> Although now that I think about it what if the user has something like this?

Such an edge case doesn't deserve the complexity of the code that a regex brings. If we find an edge case like that we will develop a PR on its own with its own unit test.

Reply:
There are thousands or millions of bugs lurking in any piece of software, you cannot expect all of them in just one line of code. We work on a 1 by 1 basis.

Reply:
makes sense. I'll bring it back